src/config_file.c

```diff
@@ -1197,7 +1244,12 @@ static int config_write(diskfile_backend *cfg, const char *orig_key, const char
 		git_buf_attach(&cfg->locked_content, git_buf_detach(&buf), len);
 	} else {
 		git_filebuf_write(&file, git_buf_cstr(&buf), git_buf_len(&buf));
-		result = git_filebuf_commit(&file);
+
+		if ((result = git_filebuf_commit(&file)) < 0)
+			goto done;
+
+		if ((result = config_refresh_from_buffer(&cfg->header.parent, buf.ptr, buf.size)) < 0)
```

Comment:
Build is still red, because Windows fails and/or leaks. I've looked around for the leak and pinpointed one possible occurrence here, but since there's also timing issues at hand, maybe not ?

`buf` was "attached" in the other code path, but not here?

Reply:
Nah, that was just a quick test of mine :P Reverted it again to the previous state, which passes